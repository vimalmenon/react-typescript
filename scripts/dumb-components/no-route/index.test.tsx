import "jsdom-global/register";
import * as React from "react";
import * as renderer from "react-test-renderer";
import NoRoute from "./index";
import {shallow} from "enzyme";

import {
    BrowserRouter as Router,
	Route
} from "react-router-dom";


describe("NoRoute component :", () => {

    test("Snapshot renders :", () => {
        const component = renderer.create(<Router><Route component={NoRoute}/></Router>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("Component test check:", () => {
        const wrapper = shallow(<NoRoute />);
        expect(wrapper.text()).toBe("No Page Found.Go back to Home");
    });
    
});