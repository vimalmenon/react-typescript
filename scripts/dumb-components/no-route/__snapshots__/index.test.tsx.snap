// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoRoute component : Snapshot renders : 1`] = `
<section
  className="sc-bdnylx eFcNAu"
>
  <span>
    No Page Found.
  </span>
  <span>
    Go back to 
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </span>
</section>
`;
